src/Mvc/Mvc.Core/src/Infrastructure/MvcCoreMvcOptionsSetup.cs

```diff
@@ -74,6 +74,8 @@ public void Configure(MvcOptions options)
         options.ModelBinderProviders.Add(new ArrayModelBinderProvider());
         options.ModelBinderProviders.Add(new CollectionModelBinderProvider());
         options.ModelBinderProviders.Add(new ComplexObjectModelBinderProvider());
+        options.ModelBinderProviders.Add(new DateOnlyModelBinderProvider());
+        options.ModelBinderProviders.Add(new TimeOnlyModelBinderProvider());
```

Comment:
I would like to get rid of the `DateTime` and `DateTimeOffset` model binders, but I don't think it's worth potentially breaking working code. @brunolins16 What do you think?

Reply:
@halter73 I do not think most people will change those binders directly. We can remove them from 8.0, make an announcement in preview1 and see if someone is broken by it.

Reply:
As reported [here](https://github.com/dotnet/aspnetcore/issues/45338)  , it could potentially be break and was decided (I don't have the historical reason) to create specific binders for them even though the simpletypebinder supported them a long ago. That said I believe this is a worthy investigation and we don't need to delete them and instead just remove the default registration and let the binder available for manual registration.

Reply:
_**Suggestion**_ 💡: What if we add those two new `Binder` to allow users customization, without the `Providers` (no registration), and let the TryParse/Converter binder do the work by default.

Reply:
These are the `DatetimeStyles` when using the TryParse binder:

https://github.com/dotnet/aspnetcore/blob/5c62d0fbc7655de8de01c37b39b07b88ed88dc5d/src/Shared/ParameterBindingMethodCache.cs#L130-L135

Reply:
Yep, `DateTimeStyles.AllowWhiteSpaces` is what I currently have for these proposed providers. While writing them I found that the only supported styles for `TimeOnly` and `DateOnly` were the ones pertaining to white space, but I can't seem to find a doc page to back that up.

Also, do you think there's any gain/loss when TryParse is used over any of the other providers/binders? I see that TryParse goes before SimpleType.